.github/workflows/build.yml

```diff
@@ -191,3 +191,33 @@ jobs:
           path: ~/.gradle/build-scan-data
           compression-level: 9
           if-no-files-found: ignore
+
+  update-test-catalog:
+    needs: test
```

Comment:
This requires successful dependent job ... I guess it is fine as our CI is quiet stable now?

Reply:
Actually, we probably should run this even if the test job fails due to flaky tests.